src/desktop-launch

```diff
@@ -57,6 +57,10 @@ else
   needs_update=true
 fi
 
+# If the user has saves in the old $SNAP_USER_DATA/.config, move them
+# to $SNAP_USER_COMMON/.config
+mv -n $SNAP_USER_DATA/.config/* $XDG_CONFIG_HOME
```

Comment:
It'd be good to structure this so it will only be run once, rather than attempt to move files every time. That is:

* Do nothing if `$SNAP_USER_DATA/.config` doesn't exist.
* After migrating the files, delete the `$SNAP_USER_DATA/.config` directory to prevent the process repeating.

If we're deciding not to clobber files Steam might have placed in `$SNAP_USER_COMMON/.config`, then perhaps deleting the files will be fine. There will be a backup in the previous numbered directory after all.